javaparser-core/src/main/java/com/github/javaparser/JavaParser.java

```diff
@@ -86,72 +83,145 @@ public static void setDoNotAssignCommentsPreceedingEmptyLines(boolean doNotAssig
         _doNotAssignCommentsPreceedingEmptyLines = doNotAssignCommentsPreceedingEmptyLines;
     }
 
-    public static CompilationUnit parse(final InputStream in,
-                                        final String encoding) throws ParseException {
-        return parse(in,encoding,true);
-    }
-
     /**
-     * Parses the Java code contained in the {@link InputStream} and returns a
-     * {@link CompilationUnit} that represents it.
+     * Parses the Java code contained in the {@link String} and returns a
+     * {@link CompilationUnit} wrapped in a {@link ParseResult} that represents
+     * it.
      *
-     * @param in
-     *            {@link InputStream} containing Java source code
+     * @param code
+     *            {@link String} containing Java source code
      * @param encoding
      *            encoding of the source code
+     * @param considerComments
+     *            whether to parse comments and attach to nodes
      * @return CompilationUnit representing the Java source code
      * @throws ParseException
      *             if the source code has parser errors
      */
-    public static CompilationUnit parse(final InputStream in,
-                                        final String encoding, boolean considerComments) throws ParseException {
+    public static ParseResult<CompilationUnit> parse(final String code,
```

Comment:
To maintain compatibility we could have both one method with the same name and the same return type as before and a new method named "parseWithTokens" or something similar. The new method would return the ParseResult and the method with the old name ("parse") would just call the new method, take the node and throw away the tokens. What do you think? I know it could be a bit repetitive to write but in this way we would stay compatible and offer a clean interface for many users which are not interested in tokens, while giving all the information to users interested in both nodes and tokens